packages/runtime/container-runtime/src/dataStore.ts

```diff
@@ -75,14 +77,25 @@ class DataStore implements IDataStore {
             case AliasState.Aliased:
                 return this.alias === alias ? "Success" : "AlreadyAliased";
 
-            // There is no current or past alias operation for this datastore,
-            // it is safe to continue execution
-            case AliasState.None: break;
+            case AliasState.None: {
+                const existingAlias = this.pendingAliases.get(alias);
```

Comment:
Can we instead wait for this operation to resolve (similar to AliasState.Aliasing case)? Cause it may fail (let's say this data store will get another alias, freeing alias this data store tries to apply). 
It feels to always better resolve when we have all the data. That said, I wonder if here (and maybe in AliasState.Aliasing case as well) we may have kind of recursion, i.e. by the time await is satisfied, we are back to similar state (imagine that in aliasing case some other task on same machine gets to run and moves data store back to aliasing state).
It feels like the safest ben is await and simple loop back to the start of the function.

Reply:
I agree with you in principle. In this case, any way in which aliasing may fail in the other operation, will also cause this to fail with a conflict. If the other operation succeeded, this is a conflict. If it's a conflict, this is also naturally a conflict. 

I'll rethink this to make it more future proof.

Reply:
> However, in that case, this datastore is free to re-apply for another alias. I'll write a test for that.

Which actually makes it inconsistent with the existing behavior. Currently, If an alias op fails, the datastore is locked out of future alias attempts.

Reply:
The inconsistency has been fixed by allowing datastores to be re-aliased after they fail the first time. Not convinced this is the right way, I'm still thinking about it, I don't see a scenario where something like that would be desired.